lib/action_policy/policy/aliases.rb

```diff
@@ -29,8 +29,15 @@ def included(base)
       end
 
       def resolve_rule(activity)
-        return activity if respond_to?(activity)
-        self.class.lookup_alias(activity) || super
+        if self.class.instance_methods(false).include?(activity)
```

Comment:
Looks a little bit complicated, especially this superclass checking. And I'm worried about the performance. A quick benchmark shows:

```ruby
require "benchmark"

GC.disable

class A
  def foo
    "foo"
  end
end

N = 1_000_000

a = A.new

Benchmark.bm(45) do |x|
  x.report("instance_methods") do
    N.times { a.class.instance_methods(false).include?(:foo) }
  end

  x.report("respond_to?") do
    N.times { a.respond_to?(:foo) }
  end
end
``` 

```
                                                  user     system      total        real
instance_methods                              0.596636   0.021402   0.618038 (  0.620872)
respond_to?                                   0.099455   0.000392   0.099847 (  0.100213)
```